### PR TITLE
added FAQ page to the website

### PR DESCRIPTION
**Pull request by dotslashbit**
Fix : (https://github.com/roc-lang/roc/issues/6260)

### CODE REVIEW

File: crates/compiler/builtins/roc/Bool.roc [hunk 1]
```diff
@@ -6,7 +6,7 @@ interface Bool
 ##
 ## Total equality means that all values of the type can be compared to each
 ## other, and two values `a`, `b` are identical if and only if `isEq a b` is
-## `Bool.true`.
+## `Bool.true`.test
```
Comment (lukewilliamboswell) [inline crates/compiler/builtins/roc/Bool.roc:9]: Is this meant to be here?

File: www/content/faq.md [hunk 2]
```diff
@@ -0,0 +1,447 @@
+# faq
```
Comment (lukewilliamboswell) [inline www/content/faq.md:1]: Can we expand this to the full Frequently Asked Questions

File: www/content/faq.md [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+
+## Where did the name Roc come from?
+
+<img width="128" alt="The Roc logo, an origami bird" src="https://user-images.githubusercontent.com/1094080/92188927-e61ebd00-ee2b-11ea-97ef-2fc88e0094b0.png">
```
Comment (lukewilliamboswell) [inline www/content/faq.md:5]: Not sure we want to load images from githubusercontent.com, we should have the svg somewhere we can load include in the static assets instead. Or at the very least include the PNG in our static assets. 

Also, I think this looks better if we move it down under the sentence "The Roc programming language is named after [a mythical bird", where it is messes with the content layout a bit.

File: www/content/faq.md [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+passing `\x -> x + 1` to a function compared to defining `fn = \x -> x + 1` elsewhere and then passing `fn` into
+the function might give different answers.
+
+Both of these would make revising code riskier across the entire language, which is very undesirable.
+
+Another option would be to define that function equality always returns `false`. So both of these would evaluate
+to `false`:
+
+- `(\x -> x + 1) == (\x -> 1 + x)`
+- `(\x -> x + 1) == (\x -> x + 1)`
```
Comment (lukewilliamboswell) [inline www/content/faq.md:57]: We could use roc syntax highlighting here now, so something like (note I've used `'` to stop it from rendering in this comment).
'''roc
(\x -> x + 1) == (\x -> 1 + x) # false
(\x -> x + 1) == (\x -> x + 1) # false
'''

File: www/content/faq.md [hunk 5]
```diff
@@ -0,0 +1,447 @@
...
+In Roc, both of these expressions evaluate to `"Hello, World!"`
+
+```elixir
+Str.concat "Hello, " "World!"
+```
+
+```elixir
+"Hello, "
+|> Str.concat "World!"
+```
```
Comment (lukewilliamboswell) [inline www/content/faq.md:260]: Change this from elixir to roc to get syntax highlighting

File: www/content/faq.md [hunk 6]
```diff
@@ -0,0 +1,447 @@
...
+
+```elixir
+someNumber
+|> Num.div 2
+```
+
+```elixir
+someNumber
+|> Num.sub 1
+```
```
Comment (lukewilliamboswell) [inline www/content/faq.md:282]: Same here - change to roc

File: www/content/faq.md [hunk 7]
```diff
@@ -0,0 +1,447 @@
...
+    someFunction
+        "some argument"
+        num
+        anotherArg
+```
+
+```elixir
+numbers
+|> List.map Num.abs
+```
```
Comment (lukewilliamboswell) [inline www/content/faq.md:304]: And here

File: www/content/faq.md [hunk 8]
```diff
@@ -0,0 +1,447 @@
...
+answer =
+    List.map
+        (\num ->
+            someFunction
+                "some argument"
+                num
+                anotherArg
+        )
+        numbers
+```
```
Comment (lukewilliamboswell) [inline www/content/faq.md:332]: I'm going to stop with this comment... there's a few more.  :smiling

File: www/content/faq.md [hunk 9]
```diff
@@ -0,0 +1,447 @@
...
+1. We struggled to get Rust to emit LLVM bitcode in the format we needed, which is important so that LLVM can do whole-program optimizations across the standard library and compiled application.
+2. Since the standard library has to interact with raw generated machine code (or LLVM bitcode), the Rust code unavoidably needed `unsafe` annotations all over the place. This made one of Rust's biggest selling points inapplicable in this particular use case.
+3. Given that Rust's main selling points are inapplicable (its package ecosystem being another), Zig's much faster compile times are a welcome benefit.
+4. Zig has more tools for working in a memory-unsafe environment, such as reporting memory leaks in tests. These have been helpful in finding bugs that are out of scope for safe Rust.
+
+The split of Rust for the compiler and Zig for the standard library has worked well so far, and there are no plans to change it.
+
+## Why is the website so basic?
+
+We have a very basic website on purpose, it helps set expectations that roc is a work in progress and not ready yet for a first release.
```
Comment (lukewilliamboswell) [inline www/content/faq.md:447]: Remove - no longer relevant

**Review by lukewilliamboswell — DISMISSED** [review]
Looking great so far, 👍 thank you for doing this @dotslashbit

File: crates/compiler/builtins/roc/Bool.roc (hunk 1 above)
Comment (dotslashbit) [reply crates/compiler/builtins/roc/Bool.roc:9]: Oh, sorry I was working on the CI issue and testing if the action is triggering or not? I'll remove it and update the PR

File: www/content/faq.md (hunk 2 above)
Comment (dotslashbit) [reply www/content/faq.md:1]: Ok, will do that now

File: www/content/faq.md (hunk 4 above)
Comment (dotslashbit) [reply www/content/faq.md:57]: Ok, working on updating the rest of the formatting

**Comment by dotslashbit** [comment]
I have made the required changes, please check and inform me if there's any correction to be done

**Review by Anton-4 — CHANGES_REQUESTED** [review]
Excellent @dotslashbit, could you just remove the current FAQ.md file at the root of the repo? We don't want the files to start drifting.

**Comment by dotslashbit** [comment]
Done, please check and let me know, thanks

**Review by Anton-4 — APPROVED** [review]
Thanks @dotslashbit!

**Comment by Anton-4** [comment]
Can you merge this @rtfeldman?

**Comment by Anton-4** [comment]
Can you set up [commit signing](https://github.com/roc-lang/roc/blob/main/CONTRIBUTING.md#commit-signing) for future contributions @dotslashbit?

**Comment by dotslashbit** [comment]
Ok I'll do that

**Comment by hoosierEE** [comment]
There is currently a [GitHub wiki entry for FAQ](https://github.com/roc-lang/roc/blob/main/FAQ.md), but it returns 404. Either removing it from the GitHub wiki or adding a link to the website [FAQ](https://www.roc-lang.org/faq.html) would be an improvement.

**Comment by Anton-4** [comment]
Thanks for alerting us @hoosierEE, I fixed it :)

**Pull request merged**